xls/modules/fp_fast_rsqrt_32.x

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file implements floating point fast (approximate) 
+// inverse square root.  Note:
+//  - Input denormals are treated as/flushed to 0.
+//      (denormals-are-zero / DAZ).
+//  - Only round-to-nearest mode is supported.
+//  - No exception flags are raised/reported.
+//  - We emit a single, canonical representation for 
+//      NaN (qnan) but accept all NaN respresentations 
+//      as input
+
+import float32
+import xls.modules.fpadd_2x32
+import xls.modules.fpmul_2x32
+import xls.dslx.stdlib.apfloat
+
+type APFloat = apfloat::APFloat;
+type F32 = float32::F32;
+
+// Determines if the given value is 0, taking into account
+// flushing subnormals.
+fn is_zero<EXP_SZ: u32, SFD_SZ: u32>(x: APFloat<EXP_SZ, SFD_SZ>) -> u1 {
```

Comment:
Seems like a general-purpose float thing - move to a standard lib?

Reply:
Done.